OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -303,17 +323,48 @@ void onFailure(int statusCode, String response, Throwable throwable) {
         }
     }
 
+    /**
+     * In order to an IAM to be re display, two condition need to be satisfied
+     *  - Gap between displays
+     *  - Quantity of displays
+     *
+     * For re display, the message need to be removed from the arrays that track the display
+     * */
+    private void setDataForReDisplay(OSInAppMessage message) {
+        if (!message.isRedisplayEnabled())
+            return;
+
+        int index = displayedMessagesData.indexOf(message);
+        if (index != -1) {
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "setDataForReDisplay: " + message.messageId);
+
+            OSInAppMessage savedIAM = displayedMessagesData.get(index);
+            message.setLastDisplayTime(savedIAM.getLastDisplayTime());
+            message.setDisplayQuantity(savedIAM.getDisplayQuantity());
+
+            if (message.isDelayTimeSatisfied() && message.shouldDisplayAgain()) {
```

Comment:
Here we need to add a check to only re-display if a trigger was updated that is in this IAM's trigger set.
This prevents the IAM from reshowing over and over until it hits it's display limit since `evaluateInAppMessages` is called each time an IAM is dismissed.